packages/aws-rfdk/lib/lambdas/lambda-layer-version-arn-mapping.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ILayerVersion,
+  LayerVersion,
+} from '@aws-cdk/aws-lambda';
+import {
+  CfnMapping,
+  Construct,
+  Stack,
+} from '@aws-cdk/core';
+import {
+  ARNS,
+} from './lambdaLayerVersionArns';
+
+/**
+ * Enumeration of our lambda layers.
+ *
+ * The values of these must match the names our lambda layers with non-alphanumeric
+ * characters removed and in lowercase.
+ */
+export enum LambdaLayer {
+  OPEN_SSL_AL2 = 'opensslal2',
+}
+
+type RegionMap<T> = { [region: string]: T };
+type LambdaLayerVersionArnMap<T> = { [layer: string]: T };
+
+/**
+ * A `CfnMapping` of our lambda layer version ARNs, from the autogenerated `lambdaLayerVersionArns.ts` file.
+ */
+export class LambdaLayerVersionArnMapping extends CfnMapping {
+  /**
+   * Gets the singleton instance (per-stack) of this class.
+   * @param scope The parent construct.
+   */
+  public static getSingletonInstance(scope: Construct) {
+    const uuid = '7a9adb59-d5d0-42fb-bc5b-a55bcbf6a6b2';
+    const uniqueId = 'LambdaLayerVersionArnMapping' + uuid.replace(/[-]/g, '');
+    const stack = Stack.of(scope);
+
+    return (stack.node.tryFindChild(uniqueId) as LambdaLayerVersionArnMapping) ?? new LambdaLayerVersionArnMapping(stack, uniqueId, ARNS);
+  }
```

Comment:
Can this be made `private` or `protected`?

Seems like we only need a static `getLambdaLayerVersion` method that could implicitly create the singleton. Then the API simply becomes:

```ts
const lambdaLayerVersion = LambdaLayerVersionArnMapping.getLambdaLayerVersion(this, 'LambdaLayerVersion', LambdaLayer.OPEN_SSL_AL2);
```

and the caller does not have to worry about calling `LambdaLayerVersionArnMapping.getSingletonInstance`

Reply:
That's much cleaner, thanks! updated

Reply:
Looks good! thanks